doc/tor.md

```diff
@@ -223,3 +223,40 @@ for normal IPv4/IPv6 communication, use:
   Otherwise it is trivial to link them, which may reduce privacy. Onion
   services created automatically (as in section 2) always have only one port
   open.
+
+- In some regions of the world the Tor network is monitored or blocked.
+  If you are in an environment that does not permit direct Tor connections or the use
+  of Tor bridges, then considering the trade-offs, it may not be safe to use Tor.
+
+- For maximum privacy, it is preferable to disable accepting incoming connections.
```

Comment:
This seems like bad advice. Incoming connections should be encouraged, and any potential privacy risks addressed.

Reply:
Bitcoin Core behind a tor proxy with no incoming connections is better for privacy in comparison to listening over Tor IMO. It was suggested by @gmaxwell and reviewers(Sipa and Jonatack) in previous PR had no issues with it:

https://github.com/bitcoin/bitcoin/issues/17491#issuecomment-705485718
https://github.com/bitcoin/bitcoin/pull/21157#discussion_r574895414

Reply:
Both of these appear to be talking about security, not privacy. And not as general advice, but as super-paranoia.

Reply:
This comment by Sipa specifically mentioned privacy: https://github.com/bitcoin/bitcoin/pull/21157#discussion_r574897060

Reply:
re: not encouraging this setting; I think the section on Sybil Attacks and Network Partitioning covers that, right?

Reply:
I think many users will read this and go "oh good, I want maximal privacy, so I'll disable listening."  IMO the statement as it is, and placed where it is, lacks nuance and context. Privacy is a slippery and multifaceted thing with a number of tradeoffs; I'm not sure that discussion should be in this doc, but it probably should be if advice like this is given in the repository, as any such advice here might be seen as a reference.

Reply:
Also, some of the thoughts here may also pertain to other privacy networks, e.g. I2P and perhaps others at some point like CJDNS.  Should there be privacy recommendations in each doc, or rather in one place (`doc/privacy.md`?) that interrelates between the networks.

Reply:
> IMO the statement as it is, and placed where it is, lacks nuance and context

Agree

> I'm not sure that discussion should be in this doc

1. This doc had a section for privacy recommendations 
2. I am trying to share few things that I learnt during different conversations about private, relevant to this doc and can help others 

> as any such advice here might be seen as a reference

They should be used as reference. I had two options when I started researching about these things:

1. Share my thoughts and discuss on social media about it. Few people will read it and forget it.
2. Create a pull request with basic things that can be added, discuss with PR reviewers and help more users know about it in a better way.

I think 2 works better.

Example: Deleting `onion_private_key` is mentioned on multiple websites: Bitcoin Wiki, Bisq Wiki etc. Editing every wiki, commenting on each social media post, etc. will take lot of time and people will still believe what they read in docs and not me.

> Should there be privacy recommendations in each doc, or rather in one place (doc/privacy.md?) that interrelates between the networks.

I think i2p and Tor have few similarities and differences. It's better we have separate docs and maintain privacy recommendations separately but have no issues with `doc/privacy.md` suggestion. I have added few things in a different repository used by lot of people to learn Bitcoin from command line:

Differences: https://github.com/prayank23/Learning-Bitcoin-from-the-Command-Line/blob/i2p/15_0_Using_i2p.md

Trade-offs: https://github.com/prayank23/Learning-Bitcoin-from-the-Command-Line/blob/i2p/15_1_i2p_service.md